### PR TITLE
SRE-66 nixify cardano-explorer-app

### PR DESCRIPTION
**Pull request by craigem**

### CODE REVIEW

File: nix/packages.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{ mkYarnPackage, stdenv, lib, python, nodejs, cardano-explorer-app-src }:
+
+{
+  cardano-explorer-app = mkYarnPackage {
+    pname = "cardano-explorer-app";
```
Comment (rhyslbw) [inline nix/packages.nix:5]: Ideally the name and version should be sourced from [package.json](https://github.com/input-output-hk/cardano-explorer-app/blob/develop/package.json)

Comment (craigem) [reply nix/packages.nix]: resolved with commit 09be18d

**Comment by craigem** [comment]
> @cleverca22 What was wrong with the `yarn.lock`? What did you do to "fix" it?

Added the SHA. `yarn2nix` is particular about that existing.

**Comment by rhyslbw** [comment]
What needs to be done here @craigem ?

File: nix/packages.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      mkdir -p $HOME/.node-gyp/${nodejs.version}
+      echo 9 > $HOME/.node-gyp/${nodejs.version}/installVersion
+      ln -sfv ${nodejs}/include $HOME/.node-gyp/${nodejs.version}
+    '';
+
+    installPhase = ''
+      sed -i 's@schema: ./node_modules@schema: ../../node_modules@' deps/cardano-explorer-app/codegen.yml
+
+      export PATH="$PATH:$node_modules/.bin"
+      yarn --offline run build
```
Comment (rhyslbw) [inline nix/packages.nix:22]:
```suggestion
      yarn install --offline --prod && yarn build 
```

Comment (rhyslbw) [reply nix/packages.nix:22]: Disregard this. Clearly `devDependencies` are required for building

**Comment by disassembler** [comment]
bors ping

**Comment by disassembler** [comment]
bors ping

**Pull request merged**